backend-rust/src/bin/ccdscan-api.rs

```diff
@@ -60,6 +64,9 @@ async fn main() -> anyhow::Result<()> {
         .connect(&cli.database_url)
         .await
         .context("Failed constructing database connection pool")?;
+    // Ensure the database schema is compatible with supported schema version.
+    migrations::ensure_compatible_schema_version(&pool, SUPPORTED_SCHEMA_VERSION).await?;
```

Comment:
Just do the same migration here instead.

During it this way we have an annoying situation where we are dependent on the api is initialising after the migration has occurred on the indexer which is a pretty annoying feature to implement deployment wise. 

At least it cannot be done as is, since it happens application side.

Reply:
I really don't think this should be doing migrations.

Most database migrations require changes to the indexer, so if the API starts doing migrations, it might crash it.
New columns need to be populated by the indexer, before the API should start using them anyway.
Removing columns would crash the indexer.

We also don't want several migrations to be running at once, so since we only run one indexer instance, this is probably a better fit.

Reply:
Currently you are not making changes to the indexer. This is database relevant only. 

As is it will cause failures to the deployment because the SQL schema will be incompatible. The simple solution is to do what I suggested above.

An alternative could be to provide the information about whether the api schema has been applied, and then provide such information. This could for example be in the health endpoint.

Reply:
A simpler approach yet is to just keep retrying until it is valid

Reply:
This was discussed offline, so resolving this.
Let me know if you disagree